Kernel/Graphics/VirtIOGPU/VirtIOGPU.cpp

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (c) 2021, Sahan Fernando <sahan.h.fernando@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Graphics/VirtIOGPU/VirtIOGPU.h>
+#include <LibC/sys/ioctl_numbers.h>
+
+#define DEVICE_EVENTS_READ 0x0
+#define DEVICE_EVENTS_CLEAR 0x4
+#define DEVICE_NUM_SCANOUTS 0x8
+
+namespace Kernel::Graphics {
+
+VirtIOGPU::VirtIOGPU(PCI::Address address)
+    : VirtIODevice(address, "VirtIOGPU")
+    , m_scratch_space(MM.allocate_contiguous_kernel_region(8 * PAGE_SIZE, "VirtGPU Scratch Space", Region::Access::Read | Region::Access::Write))
+{
+    VERIFY(!!m_scratch_space);
+    dbgln("Gonna configure now");
```

Comment:
did you mean to leave this in?

Reply:
No actually, I'll remove this in the next revision. :)